ykrt/src/frame/mod.rs

```diff
@@ -33,7 +46,18 @@ pub static AOT_STACKMAPS: LazyLock<Vec<SMEntry>> = LazyLock::new(|| {
             usize::try_from(sec.size()).unwrap(),
         )
     };
-    StackMapParser::get_entries(slice)
+    let (entries, numrecs) = StackMapParser::get_entries(slice);
+    let mut pinfos = Vec::new();
+    let mut records = Vec::new();
+    records.resize_with(usize::try_from(numrecs).unwrap(), || (Record::empty(), 0));
+    for entry in entries {
+        pinfos.push(entry.pinfo);
+        for r in entry.records {
+            let idx = usize::try_from(r.id).unwrap();
+            records[idx] = (r, pinfos.len() - 1);
```

Comment:
Are we sure that `pinfos.len()` is bigger than 0 at this point?

Reply:
I checked and the only way for it to be `0` is if there's not a single stackmap in the program. Which in turn would require the program to have no branches. But that would be a pretty silly program and since it can't have loops we can't trace anything so will never end up in deopt.

Reply:
OK so I suggest:

1. adding a check to the trace compiler that this doesn't happen (i.e. borking in the trace compiler if someone tries to compile such a trace),
2. adding a comment here referencing the check in the compiler.

Note: *maybe* we can even detect this AOT and not at trace creation time? I dunno.

Reply:
Wait, I got an even better answer. We literally push to `pinfos` just before this loop. So it's guaranteed to always have at least one entry.

Reply:
Aha!